econml/utilities.py

```diff
@@ -988,379 +988,6 @@ def _safe_norm_ppf(q, loc=0, scale=1):
-
-        if sample_var is None:
-            if np.any(np.not_equal(sample_weight, 1)):
-                warnings.warn(
-                    "No variance information was given for samples with sample_weight not equal to 1, "
-                    "that represent summaries of multiple original samples. Inference will be invalid!")
-            sample_var = np.zeros(y.shape)
-
-        if sample_var.ndim < 2:
-            if np.any(np.equal(sample_weight, 1) & np.not_equal(sample_var, 0)):
-                warnings.warn(
-                    "Variance was set to non-zero for an observation with sample_weight=1! "
-                    "sample_var represents the variance of the original observations that are "
-                    "summarized in this sample. Hence, cannot have a non-zero variance if only "
-                    "one observations was summarized. Inference will be invalid!")
-        else:
-            if np.any(np.equal(sample_weight, 1) & np.not_equal(np.sum(sample_var, axis=1), 0)):
-                warnings.warn(
-                    "Variance was set to non-zero for an observation with sample_weight=1! "
-                    "sample_var represents the variance of the original observations that are "
-                    "summarized in this sample. Hence, cannot have a non-zero variance if only "
-                    "one observations was summarized. Inference will be invalid!")
-
-        if X is None:
-            X = np.empty((y.shape[0], 0))
-
-        assert (X.shape[0] == y.shape[0] ==
-                sample_weight.shape[0] == sample_var.shape[0]), "Input lengths not compatible!"
-        if y.ndim >= 2:
-            assert (y.ndim == sample_var.ndim and
-                    y.shape[1] == sample_var.shape[1]), "Input shapes not compatible: {}, {}!".format(
-                y.shape, sample_var.shape)
-
-        return X, y, sample_weight, sample_var
-
-    def fit(self, X, y, sample_weight=None, sample_var=None):
-        """
-        Fits the model.
-
-        Parameters
-        ----------
-        X : (N, d) nd array like
-            co-variates
-        y : {(N,), (N, p)} nd array like
-            output variable(s)
-        sample_weight : (N,) nd array like of integers
-            Weight for the observation. Observation i is treated as the mean
-            outcome of sample_weight[i] independent observations
-        sample_var : {(N,), (N, p)} nd array like
-            Variance of the outcome(s) of the original sample_weight[i] observations
-            that were used to compute the mean outcome represented by observation i.
-
-        Returns
-        -------
-        self : StatsModelsLinearRegression
-        """
-        # TODO: Add other types of covariance estimation (e.g. Newey-West (HAC), HC2, HC3)
-        X, y, sample_weight, sample_var = self._check_input(X, y, sample_weight, sample_var)
-
-        if self.fit_intercept:
-            X = add_constant(X, has_constant='add')
```

Comment:
Since you also clean imports in this PR, this import could also be deleted

Reply:
Sorry, it's not clear to me what you're referring to here - could you explain?

Reply:
ohh sorry I should point to the function name, `add_constant` can be deleted from import.

Reply:
I removed it, thanks for pointing it out.